app/modules/websocket.c

```diff
@@ -183,8 +185,59 @@ static int websocketclient_connect(lua_State *L) {
   data->self_ref = luaL_ref(L, LUA_REGISTRYINDEX);
 
   const char *url = luaL_checkstring(L, 2);
-  const char *extraHeaders = luaL_optstring(L, 3, NULL);
-  ws_connect(ws, url, extraHeaders);
+  ws_connect(ws, url);
+
+  return 0;
+}
+
+static header_t *realloc_headers(header_t *headers, int new_size) {
+  if(headers) {
+    for(header_t *header = headers; header->key; header++) {
+      c_free(header->value);
+      c_free(header->key);
+    }
+    c_free(headers);
+  }
+  if(!new_size)
+    return NULL;
+  return (header_t *)c_malloc(sizeof(header_t) * (new_size + 1));
+}
+
+static int websocketclient_config(lua_State *L) {
+  NODE_DBG("websocketclient_config is called.\n");
+
+  ws_info *ws = (ws_info *) luaL_checkudata(L, 1, METATABLE_WSCLIENT);
+  luaL_argcheck(L, ws, 1, "Client websocket expected");
```

Comment:
You are right - luaL_checkudata throws an error, I've removed following luaL_argcheck call (for this and 4 other similar occurrences)